### PR TITLE
Add web startup analyzer to material 3 demo

### PR DESCRIPTION
**Pull request by johnpryan**
This adds a tool to measure web app startup for the Material 3 demo.

Demo:
- [Example app](https://flutter-web-perf-experiments.web.app/)
- [Material 3](https://flutter-web-perf-experiments--material3-vswzldcy.web.app/) (open console)

### CODE REVIEW

**Review by domesticmouse — APPROVED** [review]
LGTM if @QuncCccccc approves

File: experimental/material_3_demo/lib/main.dart [hunk 1]
```diff
@@ -2,14 +2,33 @@
...
 import 'package:flutter/material.dart';
+import 'package:web_startup_analyzer/web_startup_analyzer.dart';
 
 import 'constants.dart';
 import 'home.dart';
 
-void main() {
+main() async {
+  var analyzer = WebStartupAnalyzer(additionalFrameCount: 10);
+  debugPrint(json.encode(analyzer.startupTiming));
```
[the remark this answers is not in the document]
Comment (johnpryan) [reply experimental/material_3_demo/lib/main.dart:15]: It looks like debugPrint still prints to the console: https://flutter-web-perf-experiments--material3-vswzldcy.web.app/

[the remark this answers is not in the document]
Comment (domesticmouse) [reply experimental/material_3_demo/lib/main.dart:15]: Did I break something?

Screenshot 2024-01-24 at 11 45 02

[the remark this answers is not in the document]
Comment (johnpryan) [reply experimental/material_3_demo/lib/main.dart:15]: No, that looks like a bug.

File: web/_packages/web_startup_analyzer/lib/web_startup_analyzer.dart [hunk 2]
```diff
@@ -22,19 +23,26 @@ class WebStartupAnalyzer {
   ValueNotifier<List<int>?> onAdditionalFrames = ValueNotifier(null);
 
   double get domContentLoaded =>
-      flutterWebStartupAnalyzer.timings['domContentLoaded'] as double;
+      (flutterWebStartupAnalyzer.timings['domContentLoaded'] as JSNumber)
```
Comment (kevmoo) [inline web/_packages/web_startup_analyzer/lib/web_startup_analyzer.dart:26]: @johnpryan – making everything wasm-friendly 😄

Comment (johnpryan) [reply web/_packages/web_startup_analyzer/lib/web_startup_analyzer.dart:26]: Thanks!

**Comment by johnpryan** [comment]
@kevmoo I was able to get the sample working using the stable channel, so 5f2f0d3ceeea1322bfd176ad07597fa04f708719 should do the trick. PTAL

**Review by QuncCccccc — APPROVED** [review]
Sorry for the late response. LGTM! Thanks!

**Review by kevmoo — APPROVED** [review]
Awesome!

**Comment by mraleph** [comment]
I don't see any information reported for `additionalFrames` with this change. When I try to debug I notice that timings callback does not seem to be firing? I did not try to debug the `web_ui` itself.

/cc @yjbanov @eyebrowsoffire - do you know why `addTimingsCallback` would not fire on Wasm+SkWasm?

**Comment by mraleph** [comment]
Okay I can see that SkWasm does not actually invoke `frameTimingsOnRasterFinish` from its `renderScene`. any ideas @eyebrowsoffire?

**Comment by eyebrowsoffire** [comment]
Yep, I need to implement this. See
https://github.com/flutter/flutter/issues/140429

- Jackson

**Pull request merged**